Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -342,20 +368,29 @@ extension ActivityStreamPanel {
                 withUserData: userData,
                 toApplication: UIApplication.sharedApplication())
         })
-
+        
+        let removeAction = ActionOverlayTableViewAction(title: NSLocalizedString("Remove", comment: "Label for \"Remove\" button"), iconString: "action_remove", handler: { action in
+            self.topSitesManager.collectionView(collectionView!, deleteItemAtIndexPath: indexPath!)
```

Comment:
Actually. You should be able to avoid this. If you look at `deleteItemAtIndexPath` you can see that it is calling `self.deleteItemHandler?(contentItem.tileURL)` Which is a handler defined on `ActivityStreamPanel` called `self.hideURLFromTopSites(url)` You should be able to just call that directly if I'm not mistaken. Then you wouldn't need to use a reference to the collectionview!